$NetBSD: patch-Makefile,v 1.2 2023/01/16 04:15:37 dholland Exp $

- don't issue -fast on solaris; it (now?) fails

- patch for reproducibility (SOURCE_DATE_EPOCH), sent upstream.

--- Makefile.orig	2021-12-16 18:43:18.000000000 +0000
+++ Makefile
@@ -92,7 +92,7 @@ endif
 
 ifeq (sunos,$(OS))
 LDFLAGS += -lsocket
-PRODFLAGS += -fast
+#PRODFLAGS += -fast
 endif
 
 ifeq (haiku,$(OS))
@@ -127,8 +127,15 @@ k_dyn: CFLAGS += $(PRODFLAGS)
 k_dyn: src/kbuild.h $(OBJS)
 	$(CC) ${CFLAGS} $(OBJS) -rdynamic -o $@ $(LDFLAGS)
 
+DATE_FMT = +%Y-%m-%d
+ifdef SOURCE_DATE_EPOCH
+	BUILD_DATE := $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u "$(DATE_FMT)")
+else
+	BUILD_DATE := $(shell date "$(DATE_FMT)")
+endif
+
 src/kbuild.h:
-	echo "#define KBUILD_DATE \"`date +%Y-%m-%d`\"" >$@
+	echo "#define KBUILD_DATE \"$(BUILD_DATE)\"" >$@
 
 test: k_test
 
